src/cast/Cast.js

```diff
@@ -767,3 +829,75 @@ export function configureCast(cast, config) {
         return /** @type {any} */ (cast)(config)
     }
 }
+
+/**
+ * @template TStrict
+ * @template TPermissive=TStrict
+ * @param {TypeSchema} schema
+ * @param {CastConfig} config
+ * @returns {Cast<TStrict, TPermissive>}
+ */
+export function makeErgoCast(schema, config) {
+    return new CastWithErgoSFEnumsImpl(schema, config)
+}
+
+/**
+ *  overrides the mkEnumVariantConstrData and readEnumUplcData methods,
+ * detecting those enum variants that use a single field, and using a non-UPLC
+ * representation that skips the field-name in the non-uplc (non-ConstrData) representation
+ * for mkEnumVariant, this means the input data will have the nested data without the field-name
+ * in its structure.  For readEnumUplcData, this means the output data will have the nested data
+ * without the field-name in its structure.
+ * @template TStrict
+ * @template TPermissive=TStrict
+ * @implements {Cast<TStrict, TPermissive>}
+ */
+class CastWithErgoSFEnumsImpl extends CastImpl {
```

Comment:
Is another interface type needed, so that `mkEnumVariantConstrData` and `readEnumUplcData` are actually accessible externally?

Reply:
I don't have any real case for exposing those methods but I'm open to that if you think it's best.